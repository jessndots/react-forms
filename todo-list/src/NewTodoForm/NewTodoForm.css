.NewTodoForm-Form {
  max-width: 300px;
  margin-left: 15px;
  display: flex;
  flex-flow: row wrap;
  align-items: stretch;
}

.NewTodoForm-Field {
  margin-left: 10px;
  margin-right: 10px;
  margin-bottom: 5px;
}


