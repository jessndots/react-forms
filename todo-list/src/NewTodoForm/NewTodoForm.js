import React, { useState } from "react";
import { Formik, Field, Form, ErrorMessage } from "formik"
import * as Yup from 'yup';
import './NewTodoForm.css'

/** Form for creating new box to add to BoxList
 * 
 * Has state for the text of the todo list item;
 * on submission, sends {todoText} to fn rec'd from parent
 */

const NewTodoForm = ({ addTodo }) => {

    /** render form */
    return (
        <Formik
            initialValues={{ todoText: '' }}
            validationSchema={Yup.object({
                todoText: Yup.string()
                    .required('Required')
            })}
            onSubmit={(values) => {
                addTodo(values);
            }}
        >
            <Form className="NewTodoForm-Form">
                <div>
                    <label className="NewTodoForm-Label" htmlFor="todoText">Task</label>
                    <Field className="NewTodoForm-Field" id="todoText" name="todoText" type="text" />
                    <ErrorMessage name="todoText" />
                </div>
                <div className="NewTodoForm-Submit">
                    <button type="submit">Add Task</button>
                </div>
            </Form>
        </Formik>

    )
}

export default NewTodoForm;