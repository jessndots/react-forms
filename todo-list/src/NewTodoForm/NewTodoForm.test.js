import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';


// Smoke test
it("renders without crashing", function() {
  render(<NewTodoForm/>)
})

// Snapshot test
it("matches screenshot", function() {
  const {asFragment} = render(<NewTodoForm/>);
  expect(asFragment()).toMatchSnapshot();
})