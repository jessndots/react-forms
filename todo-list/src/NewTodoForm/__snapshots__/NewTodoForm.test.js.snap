// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot 1`] = `
<DocumentFragment>
  <form
    action="#"
    class="NewTodoForm-Form"
  >
    <div>
      <label
        class="NewTodoForm-Label"
        for="todoText"
      >
        Task
      </label>
      <input
        class="NewTodoForm-Field"
        id="todoText"
        name="todoText"
        type="text"
        value=""
      />
    </div>
    <div
      class="NewTodoForm-Submit"
    >
      <button
        type="submit"
      >
        Add Task
      </button>
    </div>
  </form>
</DocumentFragment>
`;
